src/main/java/gyro/google/storage/BucketAccessControlResource.java

```diff
@@ -0,0 +1,294 @@
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Regex;
+import gyro.core.validation.Required;
+import gyro.core.validation.ValidStrings;
+import gyro.google.Copyable;
+import gyro.google.GoogleResource;
+
+import java.io.IOException;
+import java.util.Set;
+
+/**
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *     google::bucket 'acl-bucket'
+ *         name: 'acl-example'
+ *         location: 'US-CENTRAL1'
+ *     end
+ *
+ *     google::acl 'acl-one'
+ *        bucket: $(google::bucket acl-bucket)
+ *        entity: 'domain-brightspot.com'
+ *        role: 'OWNER'
+ *     end
+ *
+ *     google::acl 'acl-two'
+ *        bucket: $(google::bucket acl-bucket)
+ *        entity: 'domain-sample.com'
+ *        role: 'READER'
+ *     end
+ */
+@Type("acl")
+public class BucketAccessControlResource extends GoogleResource implements Copyable<BucketAccessControl> {
+    
+    private static final String ENTITY_REGEX = "\\b(allUsers|allAuthenticatedUsers)\\b|^(user|group|domain|project)-.*?";
+
+    private BucketResource bucket;
+    private String role;
+    private String entity;
+    private String userProject;
+    private String id;
+    private String selfLink;
+    private String domain;
+    private String email;
+    private String entityId;
+    private BucketAccessControlProjectTeam projectTeam;
+
+    /**
+     * The associated Bucket.
+     */
+    @Required
+    public BucketResource getBucket() {
+        return bucket;
+    }
+
+    public void setBucket(BucketResource bucket) {
+        this.bucket = bucket;
+    }
+    
+    /**
+     * The access permission for the entity. Valid values are ``OWNER``, ``READER``, or ``WRITER``.
+     */
+    @Required
+    @Updatable
+    @ValidStrings({"OWNER", "READER", "WRITER"})
+    public String getRole() {
+        return role;
+    }
+
+    public void setRole(String role) {
+        this.role = role;
+    }
+
+    /**
+     * The entity holding the permission, in one of the following forms ``user-<userId>``, ``user-<email>``, ``group-<groupId>``, ``group-<email>``, ``domain-<domain>``, ``project-<team-projectId>``, ``allUsers``, or ``allAuthenticatedUsers``.
+     */
+    @Regex(ENTITY_REGEX)
+    @Required
+    @Updatable
+    public String getEntity() {
+        return entity;
+    }
+
+    public void setEntity(String entity) {
+        this.entity = entity;
+    }
+
+    /**
+     * The project to be billed for this request.
+     */
+    @Updatable
+    public String getUserProject() {
+        return userProject;
+    }
+
+    public void setUserProject(String userProject) {
+        this.userProject = userProject;
+    }
+
+    /**
+     * The domain associated with the entity.
+     */
+    public String getDomain() {
+        return domain;
+    }
+
+    public void setDomain(String domain) {
+        this.domain = domain;
+    }
+
+    /**
+     * Unique ID for the resource.
+     */
+    @Id
```

Comment:
The `@Id` annotation should be avoided until we know which field is used to reference this resource when using with other resources. For example Network Resource has the `name` annotated as the `@Id` as it is referenced using a fully qualified url having the name at the end and also the refresh uses the name to fetch the Network Resource from the cloud.

The `@Id` is used when `findById()` is used in refresh/copyFrom to try to find the resource based on the value that one gets from the cloud of a resource that references this.

Reply:
I've removed the annotation.